calico_containers/pycalico/ipam.py

```diff
@@ -92,7 +90,17 @@ def _compare_and_swap_block(self, block):
             except EtcdAlreadyExist:
                 raise CASError(str(block.cidr))
 
-    @handle_errors
+    def _delete_block(self, block):
+        """
+        Delete a block using an atomic delete operation.
+        """
+        try:
+            self.etcd_client.delete(
+                block.db_result.key,
+                prevIndex=block.db_result.modifiedIndex)
+        except EtcdAlreadyExist:
```

Comment:
Nope - I copied this from another deletion branch which was also wrong.  I tested last night - and is fixed.